src/librustc_privacy/lib.rs

```diff
@@ -795,10 +795,15 @@ impl<'a, 'tcx> Visitor<'tcx> for TypePrivacyVisitor<'a, 'tcx> {
             }
             hir::ExprMethodCall(_, span, _) => {
                 // Method calls have to be checked specially.
-                let def_id = self.tables.type_dependent_defs()[expr.hir_id].def_id();
                 self.span = span;
-                if self.tcx.type_of(def_id).visit_with(self) {
-                    return;
+                if let Some(def) = self.tables.type_dependent_defs().get(expr.hir_id) {
+                    let def_id = def.def_id();
+                    if self.tcx.type_of(def_id).visit_with(self) {
+                        return;
+                    }
+                } else {
+                    self.tcx.sess.delay_span_bug(expr.span,
+                                                 "no type-dependent def for method call");
```

Comment:
There's no differentiation between this case and https://github.com/rust-lang/rust/pull/51882/files#diff-1651ba5b47f9ffa648d59265d3f94becR1018

Reply:
I figured this wouldn't really be a problem, because if it ever does occur, the stack trace should narrow it down precisely. I'm also hoping that if this case is ever hit, the error message won't be hit anyway, because a proper UI error should have occurred.